### PR TITLE
S2024 simulation

### PR DESCRIPTION
**Pull request by hermya**
This PR contains:
Test recorder and route generator with cmd args
Test replayer with cmd args
Launch file to run test in carla
misc notes in pure-pursuit

### CODE REVIEW

**Review by krishauser — COMMENTED** [review]
I am concerned about the commented out code changing the trajectory tracker for testing in Carla. What is the problem using the standard tracker?

**Comment by hermya** [comment]
> I am concerned about the commented out code changing the trajectory tracker for testing in Carla. What is the problem  using the standard tracker?

- The route generated in CARLA doesn't have intermediate points in lane change situations.
- In our demo video, if you seek to 1:43, you'll see a break in the line (Which appears weird in the graph, but the y-axis is zoomed out heavily). That is a lane change. In pure pursuit, we look at points in the path in the limits of -5 and +5. 
- Since the point after the lane change was far away, the car would run in circles around itself before reaching the last point before the lane change. 
- Keeping it 0 to 20 does the work

**Comment by krishauser** [comment]
This should not be a problem if the trajectory was discretized finely enough (you can use existing methods for this) or the trajectory closest point query found the closest point within a segment. I do not approve this workaround.

**Comment by hermya** [comment]
> This should not be a problem if the trajectory was discretized finely enough (you can use existing methods for this) or the trajectory closest point query found the closest point within a segment. I do not approve this workaround.

I understand
I'll add a code to modify CARLA route itself (trace more points between lane shifts), so this workaround is not necessary

**Comment by hermya** [comment]
> This should not be a problem if the trajectory was discretized finely enough (you can use existing methods for this) or the trajectory closest point query found the closest point within a segment. I do not approve this workaround.

I've added code to generate internal fine points in the CARLA side, Here's the difference in how the points look when plotted:
Previous:
before
Now:
after

Please note: The routes in both images are different but the lane change scenario is similar

File: GEMstack/onboard/planning/pure_pursuit.py
Comment (krishauser) [inline GEMstack/onboard/planning/pure_pursuit.py:1]: Indentation problem

File: testing/recordings/new-5.json
Comment (krishauser) [inline testing/recordings/new-5.json:1]: These recordings need more descriptive names. It is impossible for others to understand what they mean when they are named things like “new-5”

**Review by krishauser — CHANGES_REQUESTED** [review]

File: testing/recordings/new-5.json
Comment (hermya) [reply testing/recordings/new-5.json:1]: I'll change the names for the existing recording files. Also
the names of recordings can be controlled by the cmd arg: prefix
running test_recorder.py as:
_**python test_recorder.py --prefix=new-5**_
will create a new recording json and route csv file with new-5 name.

**Comment by hermya** [comment]
I've solved the above requests, please let me know if any other changes are expected

**Review by krishauser — APPROVED** [review]
Removed temp file. Otherwise approved

**Pull request merged**